### ISSUE TITLE
missing line in st_gat.py

### ISSUE THREAD

**Issue by akashsonowal** [jswang/cs224w_traffic_prediction#1]
Hi team,

I think there is one line missing after line 73 in the st_gat.py file

In line 70, the shape of x is (50, 228, 12)
After x = torch.movedim(x, 2, 0) in line 73, the shape of x is (12, 228, 50)
So, to convert the shape of x to (12, 50, 228), there should be a line added.
x = torch.movedim(x, 2, 1)

Thanks,

Akash

**Comment by buptycz** [comment]
Hi @akashsonowal,

I was confused as you when I first saw the code, but I found out that torch.movedim() works like not swaping the dimension of index as we thought, but inserting. Like the code example,  After `x = torch.movedim(x, 2, 0)`, the shape of x is not actually (12, 228, 50), but inserting 12 (index 2, 3rd dimension) before 50 (index 0, 1st dimension) which is the exact shape as (12, 50, 228).

By the way, I did find other suspicious little bug in line 39~40 at file `st_gat.py`, where I suppose 
`self.lstm2 = torch.nn.LSTM(input_size=lstm1_hidden_size, hidden_size=lstm2_hidden_size, num_layers=1) `
`for name, param in self.lstm1.named_parameters():` 

should be (change the second line `self.lstm_index.named_parameters():`)

`self.lstm2 = torch.nn.LSTM(input_size=lstm1_hidden_size, hidden_size=lstm2_hidden_size, num_layers=1)`
`for name, param in self.lstm2.named_parameters():`

since this line of code should initialize the parameters of LSTM2 if I understand correctly.
